### PR TITLE
Set new command line interface

### PR DESCRIPTION
**Pull request by THenry14**
Related to #86 

## Introduced changes

- This PR is a part of the stack that aims to allow for raw text profiling.

## Stack:
--> Set new interface (This PR)
-- Clean up main function (https://github.com/software-mansion/cairo-profiler/pull/131)
-- Add view (raw text profiling) logic (https://github.com/software-mansion/cairo-profiler/pull/132)
-- Add tests for new functionality (https://github.com/software-mansion/cairo-profiler/pull/133)
-- Add docs (https://github.com/software-mansion/cairo-profiler/pull/134)

### CODE REVIEW

File: crates/cairo-profiler/src/cli/mod.rs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+
+pub(crate) mod build_profile;
+pub(crate) mod view;
+
+#[derive(Parser)]
+#[command(version, args_conflicts_with_subcommands = true)]
+#[clap(name = "cairo-profiler")]
+pub struct Cli {
+    #[clap(flatten)]
+    pub build_profile_args: Option<BuildProfile>,
```
Comment (ksew1) [inline crates/cairo-profiler/src/cli/mod.rs:13]:
```suggestion
    pub build_profile_args: BuildProfile,
```

File: crates/cairo-profiler/src/main.rs [hunk 2]
```diff
@@ -10,108 +10,80 @@ use crate::versioned_constants_reader::read_and_parse_versioned_constants_file;
...
+                .expect("Failed to encode the profile to the buffer");
+
+            let mut buffer_reader = buffer.reader();
+            let mut encoder = GzEncoder::new(&mut buffer_reader, Compression::default());
+
+            let mut encoded = vec![];
+            encoder
+                .read_to_end(&mut encoded)
+                .context("Failed to read bytes from the encoder")?;
+            file.write_all(&encoded).unwrap();
```
Comment (ksew1) [inline crates/cairo-profiler/src/main.rs:83]: [super nit]
```suggestion
            fs::write(&build_cli.output_path, &encoded)?;
```

This way we don't need:

```
          let mut file = fs::File::create(&build_cli.output_path)
                .context("Failed to create the output file")?;
```

File: crates/cairo-profiler/src/cli/mod.rs (hunk 1 above)
Comment (THenry14) [reply crates/cairo-profiler/src/cli/mod.rs:13]: What are the upsides of this potentially? I kinda like it being explicitly stated that this is Optional, but am ready to change my mind :D

Comment (ksew1) [reply crates/cairo-profiler/src/cli/mod.rs:13]: After some research i come to conclusion that marking this with option is better solution. I even made PR to fix this in coverage:

https://github.com/software-mansion/cairo-coverage/pull/124

**Review by ksew1 — APPROVED** [review]
LGTM!

**Review by piotmag769 — COMMENTED** [review]
👍

**Pull request merged**